admin/src/main/resources/io/buoyant/admin/js/src/delegator.js

```diff
@@ -14,29 +14,80 @@ define([
     }
 
     function renderAll(resp) {
-      $('.result').html(renderNode(resp));
+      $('.result').html(renderNode(resp, true));
     }
 
-    function renderNode(obj, weight){
+    function treeType(obj){
       switch(obj.type) {
-        case "delegate": obj.isDelegate = true; obj.child = renderNode(obj.delegate); break;
+        case "delegate": return treeType(obj.delegate);
+        case "transformation": return treeType(obj.tree);
+        case "alt":
+          for (var i = 0; i < obj.alt.length; i++) {
+            var subtreeType = treeType(obj.alt[i]);
```

Comment:
if we define subtreeType in here, of it's the case "union" etc then subtreeType isn't defined.....

```
Uncaught TypeError: subtreeType is not a function
    at http://localhost:9990/files/js/src/delegator.js:33:52
    at Array.some (native)
    at treeType (http://localhost:9990/files/js/src/delegator.js:33:25)
    at renderNode (http://localhost:9990/files/js/src/delegator.js:53:19)
    at HTMLButtonElement.renderAll (http://localhost:9990/files/js/src/delegator.js:17:25)
```